### PR TITLE
Add telemetry for begin/end request and unhandled exceptions.

### PR DESCRIPTION
**Pull request by Tratcher**
#353, #354 
@Eilon @rynowak @lodejard @davidfowl

### CODE REVIEW

**Comment by Eilon** [comment]
Looks pretty straightforward, :shipit:

File: src/Microsoft.AspNet.Hosting/WebHostBuilder.cs [hunk 1]
```diff
@@ -91,6 +92,10 @@ private IServiceCollection BuildHostingServices()
             services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
             services.AddLogging();
 
+            var telemetrySource = new TelemetryListener("Microsoft.AspNet");
+            services.AddInstance<TelemetrySource>(telemetrySource);
+            services.AddInstance<TelemetryListener>(telemetrySource);
```
Comment (rynowak) [inline src/Microsoft.AspNet.Hosting/WebHostBuilder.cs]: Looks very nice - I'll remove this bit from MVC.

**Comment by rynowak** [comment]
/cc @delmyers @avanderhoorn as FYI

**Annnnnnnnnnnnnnnnnnnnnnnnnd......................................** :shipit:

File: src/Microsoft.AspNet.Hosting/WebHostBuilder.cs (hunk 1 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Hosting/WebHostBuilder.cs]: Orthogonal to this PR, but @davidfowl - do you still think we need to do something different here?

**Comment by delmyers** [comment]
Thanks!

**Pull request merged**